src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -0,0 +1,83 @@
+﻿#include "pch.h"
+#include "Profiles.h"
+#include "Profiles.g.cpp"
+
+using namespace winrt;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::AccessCache;
+using namespace winrt::Windows::Storage::Pickers;
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    Profiles::Profiles()
+    {
+        m_profileModel = winrt::make<Model::implementation::ProfileModel>();
+        InitializeComponent();
+    }
+
+    Profiles::Profiles(Model::ProfileModel profile) :
+        m_profileModel{ profile }
+    {
+        InitializeComponent();
+    }
+
+    Model::ProfileModel Profiles::ProfileModel()
+    {
+        return m_profileModel;
+    }
+
+    fire_and_forget Profiles::BackgroundImage_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+        auto lifetime = get_strong();
+
+        FileOpenPicker picker;
+
+        picker.ViewMode(PickerViewMode::Thumbnail);
+        picker.SuggestedStartLocation(PickerLocationId::PicturesLibrary);
+        picker.FileTypeFilter().ReplaceAll({ L".jpg", L".jpeg", L".png", L".gif" });
+
+        StorageFile file = co_await picker.PickSingleFileAsync();
+        if (file != nullptr)
+        {
+            BackgroundImage().Text(file.Path());
+        }
+    }
+
+    fire_and_forget Profiles::Commandline_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+        auto lifetime = get_strong();
+
+        FileOpenPicker picker;
+
+        picker.ViewMode(PickerViewMode::Thumbnail);
+        picker.SuggestedStartLocation(PickerLocationId::ComputerFolder);
+        picker.FileTypeFilter().ReplaceAll({ L".bat", L".exe" });
```

Comment:
Batch files and EXE files is an interesting choice for command lines. There are other executable types and scripts...

Reply:
So, this one's weird. It's still a textbox so the user should be able to put in anything they want. That said, what other file extensions are supported here? I can just add them to the list.

Reply:
What about .cmd files?

Reply:
Don't do `bat` or `cmd` fies - not unless we're prepared to auto-prefix them with a `cmd.exe /c `. You can't just call `CreateProcess` on a `bat` file, it needs to be executable.

Reply:
Not blocking on this. This is going into a feature branch that will have a final review when it lands.


NOTE FOR EPIC: **MAKE SURE COMMANDLINE HANDLING IS ROBUST AND INTELLIGENT**